app/forms/finance/ecf/change_training_status_form.rb

```diff
@@ -0,0 +1,112 @@
+  module ECF
+    class ChangeTrainingStatusForm
+      include ActiveModel::Model
+      include ActiveModel::Attributes
+
+      REASON_OPTIONS = {
+        "deferred" => Participants::Defer::ECF.reasons,
+        "withdrawn" => Participants::Withdraw::ECF.reasons,
+      }.freeze
+
+      attribute :participant_profile
+      attribute :training_status
+      attribute :reason
+
+      validates :training_status, inclusion: ParticipantProfile.training_statuses.values
+      validates :reason, inclusion: { in: :valid_training_status_reasons }, if: :reason_required?
+
+      def training_status_options
+        ParticipantProfile.training_statuses.except(current_training_status)
+      end
+
+      def reason_required?
+        training_status.present? && training_status != "active"
+      end
+
+      def reason_options
+        REASON_OPTIONS.except(current_training_status)
+      end
+
+      def current_training_status
+        induction_record&.training_status
+      end
+
+      def valid_training_status_reasons
+        reason_options[training_status] || []
+      end
+
+      def participant_class_name
+        case participant_profile.participant_type
+        when :ect
+          "EarlyCareerTeacher"
+        when :mentor
+          "Mentor"
+        else
+          raise "Participant type not recognised"
+        end
+      end
+
+      def action_class_name
+        case training_status
+        when "active"
+          "Resume"
+        when "deferred"
+          "Defer"
+        when "withdrawn"
+          "Withdraw"
+        else
+          raise "training_status type not recognised"
+        end
+      end
+
+      # this is not correct because a participant may changeable induction records
+      # with different lead provider. But due to the participant drilldown not being
+      # correct we will make the assumpion that we will change it on the latest induction record
+      # regarless of the lead provider
+      def cpd_lead_provider
+        @cpd_lead_provider ||= participant_profile.induction_records.latest.lead_provider.cpd_lead_provider
+      end
+
+      def course_identifier
+        @course_identifier ||=
+          case participant_profile.participant_type
+          when :mentor
+            "ecf-mentor"
+          when :ect
+            "ecf-induction"
+          end
+      end
+
+      def save
+        return false unless valid?
+        return true if status_unchanged?
+
+        klass = "Participants::#{action_class_name}::#{participant_class_name}".constantize
+        klass.call(
+          params: {
+            cpd_lead_provider:,
+            course_identifier:,
+            participant_id: participant_profile.participant_identity.external_identifier,
+            reason:,
+            force_training_status_change: true,
+          },
+        )
+
+        true
+      end
+
+    private
+
+      def induction_record
+        @induction_record ||= participant_profile&.induction_records&.latest
```

Comment:
`participant_profile` is always going to be present. In the controller `ParticipantProfile.find` will raise an `ActiveRecord::Error::NotFound` if it can't find the profile